src/Output/UnifiedDiffOutputBuilder.php

```diff
@@ -51,75 +73,146 @@ public function getDiff(array $diff): string
+                    ;
+
+                    $this->writeHunk(
+                        $diff,
+                        $hunkCapture - $contextStartOffset,
+                        $i - $cutOff + $contextEndOffset + 1,
+                        $fromStart - $contextStartOffset,
+                        $fromRange - $cutOff + $contextStartOffset + $contextEndOffset,
+                        $toStart - $contextStartOffset,
+                        $toRange - $cutOff + $contextStartOffset + $contextEndOffset,
+                        $output
+                    );
 
                     $fromStart += $fromRange;
                     $toStart += $toRange;
+
+                    $hunkCapture = false;
+                    $sameCount   = $toRange = $fromRange = 0;
                 }
 
-                $start        = $commonEnd + 1;
-                $commonLength = $commonEnd - $commonStart + 1; // calculate number of non-change lines in the common part
-                $fromStart += $commonLength;
-                $toStart += $commonLength;
-            } while (false !== \next($old));
-
-            \end($old); // short cut for finding possible last `change entry`
-            $tmp = \key($old);
-            \reset($old);
-            if ($old[$tmp] === $upperLimit - 1) {
-                $upperLimit = $tmp;
+                continue;
             }
-        }
 
-        if ($start < $upperLimit - 1) { // check for trailing (non) diff entries
-            do {
-                --$upperLimit;
-            } while (isset($diff[$upperLimit][1]) && $diff[$upperLimit][1] === 0);
-            ++$upperLimit;
+            $sameCount = 0;
+
+            if ($entry[1] === self::$noNewlineAtOEFid) {
+                continue;
+            }
 
-            list($fromRange, $toRange) = $this->getChunkRange($diff, $start, $upperLimit);
-            $this->writeChunk($output, $diff, $start, $upperLimit, $fromStart, $fromRange, $toStart, $toRange);
+            if (false === $hunkCapture) {
+                $hunkCapture = $i;
+            }
+
+            if (1 === $entry[1]) { // added
+                ++$toRange;
+            }
+
+            if (2 === $entry[1]) { // removed
+                ++$fromRange;
+            }
+        }
+
+        if (false !== $hunkCapture) {
+            $contextStartOffset = $hunkCapture - $this->contextLines < 0
+                ? $hunkCapture
+                : $this->contextLines
+            ;
+
+            $this->writeHunk(
+                $diff,
+                $hunkCapture - $contextStartOffset,
+                \count($diff),
+                $fromStart - $contextStartOffset,
+                $fromRange + $contextStartOffset,
+                $toStart - $contextStartOffset,
+                $toRange + $contextStartOffset,
+                $output
+            );
         }
     }
 
-    private function writeChunk(
-        $output,
+    private function writeHunk(
         array $diff,
-        int $diffStartIndex,
```

Comment:
dropping the typehinting by purpose ?

Reply:
oh no, lemme get those back.. I think I backported the udiff formatter with 5.6 support which was not needed here